### PR TITLE
hdf and opengl fixes

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

**Review by ResearchDaniel — DISMISSED** [review]

File: modules/opengl/src/shader/shader.cpp [hunk 1]
```diff
@@ -169,8 +129,8 @@ Shader::Shader(const char *vertexFilename, const char *geometryFilename,
 
 Shader::Shader(const Shader &rhs) : program_{rhs.program_}, warningLevel_{rhs.warningLevel_} {
     for (auto &elem : rhs.shaderObjects_) {
-        shaderObjects_.emplace(
-            elem.first, ShaderAttachment(this, std::make_unique<ShaderObject>(elem.second.obj())));
+        shaderObjects_.emplace_back(elem);
+        attached_.emplace_back(false);
```
Comment (martinfalk) [inline modules/opengl/src/shader/shader.cpp:136]: what about the callbacks here? Isn't the invariant here that `shaderObjects_`, `attached_`, and `callbacks_` have the same lengths?

File: modules/opengl/src/shader/shader.cpp [hunk 2]
```diff
@@ -324,6 +318,13 @@ void Shader::rebuildShader(ShaderObject *obj) {
                   "Id: " + toString(program_.id) + ", resource: " + obj->getFileName() +
                       " successfully reloaded",
                   LogLevel::Info, LogAudience::User);
+                  
+        auto log = utilgl::getProgramInfoLog(program_.id);
+        if (!log.empty()) {
```
Comment (martinfalk) [inline modules/opengl/src/shader/shader.cpp:323]: might cause some logging on Intel or AMD

**Review by martinfalk — CHANGES_REQUESTED** [review]
nice refactoring of the shader objects

File: modules/opengl/src/shader/shader.cpp (hunk 1 above)
Comment (petersteneteg) [reply modules/opengl/src/shader/shader.cpp:136]: Yes, think I had changes that, but seems I lost some changes...

File: modules/opengl/src/shader/shader.cpp (hunk 2 above)
Comment (petersteneteg) [reply modules/opengl/src/shader/shader.cpp:323]: think it is better to log more then less in the case of OpenGL. If someone thinks it is to much then we can try an limit it

**Review by martinfalk — APPROVED** [review]

**Pull request merged**